src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs

```diff
@@ -117,9 +117,17 @@ internal static OtlpLogs.LogRecord ToOtlpLog(this LogRecord logRecord)
                     SeverityNumber = GetSeverityNumber(logRecord.Severity),
                 };
 
-                if (logRecord.Severity.HasValue)
+                if (!string.IsNullOrWhiteSpace(logRecord.SeverityText))
                 {
-                    otlpLogRecord.SeverityText = SeverityTextMapping[(int)logRecord.Severity.Value];
+                    otlpLogRecord.SeverityText = logRecord.SeverityText;
+                }
+                else if (logRecord.Severity.HasValue)
+                {
+                    uint severityNumber = (uint)logRecord.Severity.Value;
+                    if (severityNumber >= 0 && severityNumber < 6)
```

Comment:
```suggestion
                    if (severityNumber < 6)
```

A `uint` should always be `>= 0`. I think that first check can be removed.

Reply:
Good call! Updated